### PR TITLE
Ports Hippie Station's Locker Mech

### PR DESCRIPTION
**Pull request by datlo**
Original PR by : https://github.com/HippieStation/HippieStation/pull/7854

GREYTIDE STATIONWIDE : DRILLIN' IN SPACE

lockermech

Ports Hippie Station's Locker Mech.

Features : 

Crafted out of maintenance/stolen parts in the crafting menu. Now requires ripley boards, and is a sub-type of the Ripley. (cannot be emagged or improved with goliath plates)

Can be equipped with makeshift drill and clamp also crafted out of the menu, or other mining mech equipment, however it only has two equipment slots.
 
Has 5 cargo space. You could somehow fits 5 lockers in the locker mech. Don't question it, it's greytide bluespace tech.

Moves slower than a Ripley and does not go faster in low pressure environment

Has as much health as a Ripley but absolutely no protection at all, 20 toolbox hits or about one full laser gun will bring the mech down.

Is spaceproof, and for the most part functions like a regular ripley, if a kinda shitty one.

🆑
rscadd: Ported Hippie Station's Locker Mech, pinnacle of Assistant Engineering crafted out of spare maintenance parts.
/🆑

### CODE REVIEW

File: code/game/mecha/makeshift/lockermech.dm [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/obj/mecha/makeshift
+	desc = "A locker with stolen wires, struts, electronics and airlock servos crudely assembled into something that ressembles the functions of a mech."
+	name = "Locker Mech"
+	icon_state = "lockermech"
+	health = 200 //same health as ripley but without any of the armor/deflections making it very vulnerable
+	deflect_chance = 0
+	lights_power = 5
+	step_in = 5 //Slower than ripley
+	max_temperature = 5000
+	armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 0, acid = 0)
```
Comment (Fox-McCloud) [inline code/game/mecha/makeshift/lockermech.dm]: we don't have fire or acid armor.

Blatant copy-pasting will get you ever time.

File: code/game/mecha/makeshift/lockermech.dm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	damage_absorption = list("brute"=1,"fire"=1,"bullet"=1,"laser"=1,"energy"=1,"bomb"=1)
```
Comment (Fox-McCloud) [inline code/game/mecha/makeshift/lockermech.dm]: Spacing on this is atrocious.

File: code/game/mecha/makeshift/lockermech.dm [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	max_equip = 2
+	wreckage = null
+	var/list/cargo = list()
```
Comment (Fox-McCloud) [inline code/game/mecha/makeshift/lockermech.dm]: I thought RIPLEYs had a cargo list too. WHy isn't this a subtype?

**Comment by Fox-McCloud** [comment]
That sprite is atrocious.

File: code/game/mecha/makeshift/lockermech.dm (hunk 3 above)
Comment (Crazylemon64) [reply code/game/mecha/makeshift/lockermech.dm]: There is some shared logic to worry about, such as goliath plates or emagging to allow equipping anything

Comment (Crazylemon64) [reply code/game/mecha/makeshift/lockermech.dm]: Moving this var to the `working` subtype would be a good idea, though, and then you could move all the cargo logic from the RIPLEY to the `working` subtype

**Comment by DesolateG** [comment]
I gotta admit, it is hilarious sounding

**Comment by Aurorablade** [comment]
As funny as it is..I am weary of ghetto mechs

File: code/game/mecha/makeshift/lockermech.dm [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	deflect_chance = 0
+	lights_power = 5
+	step_in = 5 //Slower than ripley
+	max_temperature = 5000
+	armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0)
+	damage_absorption = list("brute"= 1, "fire"= 1, "bullet"= 1, "laser"= 1, "energy"= 1, "bomb"= 1)
+	max_equip = 2
+	wreckage = null
+	var/list/cargo = list()
+	var/cargo_capacity = 5
```
Comment (Crazylemon64) [inline code/game/mecha/makeshift/lockermech.dm]: The logic here seems like it would be implemented better as a `/datum/component`

Comment (datlo) [reply code/game/mecha/makeshift/lockermech.dm]: Not sure what you mean, the ripley code uses
```
var/list/cargo = new
var/cargo_capacity = 15
```
Is that it?

Comment (Fox-McCloud) [reply code/game/mecha/makeshift/lockermech.dm]: Again, why isn't this a Ripley subtype?

Comment (datlo) [reply code/game/mecha/makeshift/lockermech.dm]: While this is similar to the Ripley, I don't want the mech to be able to get goliath plates or emagged, considering most stats are different than the Ripley aswell I'm not sure it should be a subtype.

Comment (Aurorablade) [reply code/game/mecha/makeshift/lockermech.dm]: you may be able to prohibit what mech equipment it can put on it.

Comment (Fox-McCloud) [reply code/game/mecha/makeshift/lockermech.dm]: you can override things like that.

If there's heavy overlap, make something a subtype and override procs or behavior where necessary, don't copy-paste.

Comment (Crazylemon64) [reply code/game/mecha/makeshift/lockermech.dm]: A component would make adding cargo to any mech trivial.

Comment (Crazylemon64) [reply code/game/mecha/makeshift/lockermech.dm]: If doing so is beyond your ken, moving all the cargo logic up to `mecha/working`, and re-implementing the makeshift mech under the `mecha/working` path, is acceptable

**Comment by datlo** [comment]
Took you guys' feedback into account, thanks, I will close this for now. Might reopen later after some improving and cleaning up, if I get around to it. 
If anyone else wants to take a shot at their own implementation of the idea, feel free to make your own PR

**Comment by datlo** [comment]
Reopening after some changes, made the mech a subtype of the Ripley and significantly harder/costly to craft. It now also requires ripley boards and a high capacity power cell, meaning you have to steal/cooperate with cargo or robotics.

This also means it can install all the tools a ripley can, which seems logically consistent since it uses its boards, although it cannot be emagged to install weapons.

**Comment by farie82** [comment]
Looks interesting, one problem though with the new things. If you require a ripley board why not just make a ripley. Since you need robotics for the board anyway. Making it easier to get will just clog up the hallways though I think. Especially in the first weeks of release.

**Comment by datlo** [comment]
A ripley requires a lot more than just the board, which you can get from cargo and not necessarily robotics (board crate).

As you and others said, without this requirement this means too many mechs for too little effort, which is why I added it

**Comment by Plotron** [comment]
Not sure how I feel about drills on an assistant-grade Mech!

**Comment by davidchan13** [comment]
I can totally see round start miners making this or cargonians slapping one together to quickly move crates about. Makes me wish there was a Mech-Welder.

**Comment by taukausanake** [comment]
I think we need more makeshift crap like this that is vastly inferior to the real stuff. Macgyver all the things!

**Comment by Aurorablade** [comment]
i am half tempted to sptire hack the locker onto a pair of ED legs and give it a safety cone hat.

**Comment by datlo** [comment]
if you can make a better sprite for it, please do so and share! it could admittedly be improved

**Comment by Aurorablade** [comment]
https://puu.sh/Bb63j/0bb72c3f29.dmi

Didn't do a wreck sprite..will need to rename...

**Comment by Ty-Omaha** [comment]
>https://puu.sh/Bb63j/0bb72c3f29.dmi

>Didn't do a wreck sprite..will need to rename...

That just looks like a copy and paste of a locker ontop of a ripley...I think it looks worse. The locker has an angle too it instead of being square on, and it doesn't flow & blend with the rest of mech.

**Comment by datlo** [comment]
this is ready for review imo, @Fox-McCloud do you still dislike the PR after the changes? (made a subtype of ripley, needs ripley boards and about as much metal as a ripley does) if yes tell me what improvements you think it needs or if you just don't like the idea of a ghetto mech which I can respect

File: code/game/mecha/mecha.dm [hunk 5]
```diff
@@ -854,7 +854,7 @@
 
 
 /obj/mecha/emag_act(user as mob)
```
Comment (Crazylemon64) [inline code/game/mecha/mecha.dm]: wow. our legacy code is horrible.

**Comment by Kyep** [comment]
This is a bad idea for many reasons:

- The sprites look bad. Both the original, and the 'new' sprites. Frankly, I'm not sure you can make the idea of 'locker mech' look good.
- This gives assistants access to mechs. Think about that for a second. Assistants - with mecha. Even aside from the balance implications of that, well, we have the robotocist job for a reason. We don't let security, or even engineering, make their own mechs. Why on earth would we let assistants do so?
- The construction of this doesn't make sense in terms of basic physics. You can't just cobble together some bits to get a working mech. Also, in this PR it works like the TARDIS, appearing to be the size of one locker, but it can fit 5 lockers within it. What? Even its dimensions are not consistent. We're giving bluespace tech to assistants now?
- Who is this actually intended for? Anyone who has legit cause to have a drill mech, like miners, will just get a proper ripley. This has literally no use except allowing greytide to be more disruptive. Also, the fact that it requires a ripley board means that, in practice, robotics is never going to let you create one.
- This comes from Hippie station, and reflects a lower RP / less serious culture than we try to aim for here.

**Comment by datlo** [comment]
Yeah, Kyep makes good arguments. I'll close this for now as I don't see this getting merged, perhaps for the better.

Sorry greytide stationwide. Not today.

**Pull request closed without merging**